contract_sale_payment_mode/models/contract_contract.py

```diff
@@ -0,0 +1,11 @@
+from odoo import api, models
+
+
+class ContractContract(models.Model):
+    _inherit = "contract.contract"
+
+    @api.depends("partner_id", "contract_type")
+    def _compute_payment_mode_id(self):
+        # Exclude contracts that already have a payment mode
+        contracts_to_compute = self.filtered(lambda c: not c.payment_mode_id)
+        return super(ContractContract, contracts_to_compute)._compute_payment_mode_id()
```

Comment:
This compute seems strange to me:

Changing the partner on a contract might (should?) change the payment mode but here, you are prevent this behavior, is it intentional?